serialization/BSON.ksy

```diff
@@ -0,0 +1,165 @@
+  document:
+    doc: "BSON is a binary format in which zero or more ordered key/value pairs are stored as a single entity. We call this entity a document."
+    seq:
+      - id: len
+        type: s4
+        doc: "Total number of bytes comprising the document."
+      - id: fields
+        type: elements_list
+        size: len - 5
+      - id: terminator
+        contents: [0]
+  elements_list:
+    seq:
+      - id: elements
+        type: element
+        repeat: eos
+  regex:
+    seq:
+      - id: pattern
+        type: cstring
+      - id: options
+        type: cstring
+  string:
+    seq:
+      - id: len
+        type: s4
+      - id: str
+        type: str
+        encoding: UTF-8
+        size: len-1
+      - id: terminator
+        contents: [0]
+  cstring:
+    seq:
+      - id: str
+        type: strz
+        encoding: UTF-8
+        doc: "MUST NOT contain '\x00', hence it is not full UTF-8."
+  f16:
+    doc: "128-bit IEEE 754-2008 decimal floating point"
+    seq:
+      - id: str
+        type: b1
+      - id: exponent
+        type: b15
+      - id: significand_hi
+        type: b49
+      - id: significand_lo
+        type: u8
+  object_id:
+    doc: "https://docs.mongodb.com/manual/reference/method/ObjectId/"
+    seq:
+      - id: epoch_time
+        type: u4
+        doc: "seconds since the Unix epoch"
+      - id: machine_id
+        type: u3
+      - id: process_id
+        type: u2
+      - id: counter
+        type: u3
+        doc: "counter, starting with a random value."
+  db_pointer:
+    seq:
+      - id: namespace
+        type: string
+      - id: id
+        type: object_id
+  binary:
+    doc: "The BSON \"binary\" or \"BinData\" datatype is used to represent arrays of bytes. It is somewhat analogous to the Java notion of a ByteArray. BSON binary values have a subtype. This is used to indicate what kind of data is in the byte array. Subtypes from zero to 127 are predefined or reserved. Subtypes from 128-255 are user-defined."
+    seq:
+      - id: len
+        type: s4
+      - id: subtype
+        type: u1
+        enum: subtype
+      - id: content
+        size: len
+    enums:
+      subtype:
+        0x00: generic #Generic binary subtype
+        0x01: function
+        0x02: binary_old #This used to be the default subtype, but was deprecated in favor of \x00. Drivers and tools should be sure to handle \x02 appropriately. The structure of the binary data (the byte* array in the binary non-terminal) must be an int32 followed by a (byte*). The int32 is the number of bytes in the repetition.
+        0x03: uuid_old #his used to be the UUID subtype, but was deprecated in favor of \x04. Drivers and tools for languages with a native UUID type should handle \x03 appropriately.
+        0x04: uuid
+        0x05: md5
+        0x80: user_defined #\x80-\xFF "User defined" subtypes. The binary data can be anything.
+  element:
+    seq:
+      - id: type
```

Comment:
Patches to resolve that are welcome ;) At the very least, we'll need a list of keywords in all languages.